examples/speaker_diarization/speaker_diarization/utils.py

```diff
@@ -0,0 +1,244 @@
+        annotation[Segment(segment.start, segment.end)] = segment.group
+
+    return annotation
+
+
+def preprocess_text(segments: List[TimeSegment]) -> str:
+    """
+    Preprocess text by removing punctuation and combining transcriptions.
+    """
+    text = " ".join([segment.label for segment in segments])
+    text = re.sub(r"[^\w\s]", "", text.lower())
+    return text
+
+
+def create_non_overlapping_segments(
+    transcription: List[LabeledTimeSegment],
+    identity=None,
+) -> List[Tuple[float, float]]:
+    """
+    Takes in a list of TimeSegments (overlapping or not) and returns a non-overlapping segment list.
+    """
+    res = []  # [(start, end), ...]
+    transcription = sorted(transcription, key=lambda x: x.start)
+
+    for t in transcription:
+        if identity is not None and t.group != identity:
+            continue
+
+        start_time = t.start
+        end_time = t.end
+
+        start_idx = -1
+        end_idx = -1
+        for i, (s, e) in enumerate(res):
+            if s <= start_time <= e:
+                start_idx = i
+            if s <= end_time <= e:
+                end_idx = i
+
+        if start_idx == end_idx and end_idx == -1:
+            # add new segment to results
+            res.append([start_time, end_time])
+        elif start_idx != -1 and end_idx == -1:
+            # extend interval by end_idx
+            res[start_idx][1] = end_time
+        elif start_idx == -1 and end_idx != -1:
+            # extend interval by start_idx
+            res[end_idx][0] = start_time
+        elif start_idx < end_idx:
+            # join two intervals
+            res[start_idx][1] = res[end_idx][1]
+            del res[end_idx]
+        elif start_idx == end_idx:
+            # interval already exists
+            continue
+
+    return res
+
+
+def remove_intersection(inf: Tuple[float, float], gt: Tuple[float, float]) -> List[Tuple[float, float]]:
+    """
+    Takes in two intervals, GT and Inf, and removes the intersect of the two from inf.
+    """
+    inf_start, inf_end = inf
+    gt_start, gt_end = gt
+
+    if inf_end < gt_start or inf_start > gt_end:
+        return [inf]
+
+    if inf_start < gt_start and inf_end > gt_end:
+        return [[inf_start, gt_start], [gt_end, inf_end]]
+    elif inf_start >= gt_start and inf_end <= gt_end:
+        return None
+    elif inf_start < gt_start:
+        return [[inf_start, gt_start]]
+    else:
+        return [[gt_end, inf_end]]
+
+
+def generate_fp(gt: List[Tuple[float, float]], inf: List[Tuple[float, float]]) -> List[Tuple[float, float]]:
+    """
+    Generates false positive intervals between a GT and Inf interval list.
+    """
+    res = inf.copy()
+
+    i = 0
+    while i != len(res):
+        for j in range(len(gt)):
+            removed = remove_intersection(res[i], gt[j])
+
+            if removed is None:
+                del res[i]
+                i -= 1
+                break
+
+            if removed is not None:
```

Comment:
`else`? or no conditional is necessary here because we've got `break` in the `if` statement above.